src/batch_iterator.rs

```diff
@@ -129,18 +125,28 @@ where
     }
 }
 
+pub trait PointCloud: Sync {
+    type Id: ToString + Send;
+    type PointIter: Iterator<Item = Point>;
+    fn nodes_in_location(&self, query: &PointQuery) -> Vec<Self::Id>;
+    fn points_in_node(&self, query: &PointQuery, node_id: Self::Id) -> Self::PointIter;
+}
+
 /// Iterator on point batches
-pub struct BatchIterator<'a> {
-    octrees: &'a [octree::Octree],
+pub struct BatchIterator<'a, C> {
+    octrees: &'a [C],
     point_location: &'a PointQuery,
     batch_size: usize,
     num_threads: usize,
     buffer_size: usize,
 }
 
-impl<'a> BatchIterator<'a> {
+impl<'a, C> BatchIterator<'a, C>
+where
+    C: PointCloud,
+{
     pub fn new(
-        octrees: &'a [octree::Octree],
+        octrees: &'a [C],
```

Comment:
-> `point_cloud`?

Reply:
I did that, but apparently saved to the old location. Committed now.